internal/service/licensemanager/grant.go

```diff
@@ -0,0 +1,233 @@
+package licensemanager
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/licensemanager"
+	"github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/create"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
+	"github.com/hashicorp/terraform-provider-aws/internal/verify"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+const (
+	ResGrant = "Grant"
+)
+
+// @SDKResource("aws_licensemanager_grant")
+func ResourceGrant() *schema.Resource {
+	return &schema.Resource{
+		CreateWithoutTimeout: resourceGrantCreate,
+		ReadWithoutTimeout:   resourceGrantRead,
+		UpdateWithoutTimeout: resourceGrantUpdate,
+		DeleteWithoutTimeout: resourceGrantDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"allowed_operations": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MinItems: 1,
+				MaxItems: len(licensemanager.AllowedOperation_Values()),
```

Comment:
Thanks =) . I figured that it was possible for the allowed operations to change over time and that this would not require updating like a static number.